### ISSUE TITLE
Active experiments are observability-blind: no training logs, metrics, or sample outputs visible until `evo run` commits

### ISSUE THREAD

**Issue by alokwhitewolf** [evo-hq/evo#59]
For any non-trivial training job (SFT, RL, etc.), the experiment stays `pending` for the entire training duration -- could be 1-2h or more. During that window the dashboard and CLI show nothing useful:

- `evo show exp_0002` -> `status: pending`, `attempts: []`, no metrics, no traces
- Dashboard tree shows the node as `?` with no drill-down content
- The branch HEAD is still the parent's commit, so `evo diff` is empty
- No way to see in-flight training loss, sample outputs, or even confirm that training is making progress

The data exists -- `train.py` is writing to `metrics.jsonl`, the worktree has the modified code, the training script produces checkpoints under `attempts/NNN/checkpoints/` -- but evo doesn't surface any of it until `evo run` finalizes the experiment.

Currently the only way to monitor a live experiment is to SSH into the box and tail files directly:
```
ssh box 'tail -f /tmp/train_exp0002.log'
cd .evo/run_0000/worktrees/exp_0002 && cat metrics.jsonl
```

That's fine when the orchestrator is the user, terrible when it's a subagent on a remote VM that the user can only watch through the dashboard.

## Concrete observed during PostTrainBench AIME run

- exp_0002 spent 70 minutes training. For all 70 minutes the dashboard's diff panel was empty, the per-task panel was empty, and `evo show` returned `pending` with `attempts: []`.
- The orchestrator's parent process had no way to know the training was healthy vs hung without re-shelling into the box and tailing log files manually.
- exp_0001 had the same problem -- training showed nothing for ~2h, then committed in one shot.

## What would fix this

Two non-mutually-exclusive directions:

1. **Stream the in-flight worktree state into evo's storage during training.**
   - When `evo run` starts, immediately register an `attempts/NNN/` directory and mark the experiment `active`.
   - Live-tail `metrics.jsonl` and stdout/stderr into `attempts/NNN/benchmark.log` while the training process runs.
   - Surface those logs in the dashboard with a tail-style view that polls every few seconds.

2. **Make the uncommitted-but-modified worktree introspectable.**
   - `evo diff exp_0002` should show the working-tree diff against parent even when nothing is committed yet (with a banner that says "uncommitted").
   - Dashboard should render the worktree's `git diff` for active experiments, not require a commit.

The first fixes "can I tell if it's healthy"; the second fixes "what is the agent actually trying".

## Adjacent issues

- #57 (discarded experiments are observability-blind) -- same shape: missing data in the dashboard for a state that should be inspectable
- #51 (attempts/<n>/diff.patch written as 0 bytes despite real commits) -- related: the commit pipeline already has a gap in capturing diffs
